[CS2113T-F08-4] UniTracker

BlubberMonster opened the pull request:
Our product is user-specific as it allows users to customize and personalize their own module information and deadlines to suit their needs. Our product allows users to organize their own daily schedules and keep track of their deadlines all on one platform.

s-t-e-f commented on docs/DeveloperGuide.md:
````diff
@@ -1,34 +1,315 @@
+Returning to module information menu...
+  ```
+Given below is the sequence diagram for the feature:  
+![Diagram](diagrams/addNewReview.png)  
````
Maybe can change the format of objects to "name :Class1" or "  :Class1"
![image](https://user-images.githubusercontent.com/65395025/113081624-9ceceb00-920b-11eb-9694-89c9540d66b1.png)

jovanhuang replied:
Generally, quite a detailed sequence diagram. The logic flows correctly, from top to bottom.

It would be better if the "ModuleInfo" is ":ModuleInfo" instead. I see the same thing in other places.

s-t-e-f replied:
Maybe can consider breaking this sequence diagram into smaller fragments using ref frames

s-t-e-f commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,315 @@
+5) `Message` <br> Example: `Good job!`
+
+Given below is the sequence diagram for the feature:  
+![Diagram](diagrams/addNewTask.png)
```
Maybe can end the lifelines of ModuleInfo and Module objects after they have been deleted 
![image](https://user-images.githubusercontent.com/65395025/113082015-5350d000-920c-11eb-8580-4b868f393ac2.png)

s-t-e-f replied:
To reduce clutter, maybe some of the activation bars and return arrows can be omitted

jovanhuang replied:
It would be good if the "printing error" methods are from the Logger instead of Ui class (same as the sequence diagram of addNewReview) for consistency. :)

jovanhuang commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,315 @@
+### 4.05 Add Zoom Link
+
+Given below is the sequence diagram for the feature:  
+![Diagram](diagrams/addZoomLink.png)
```
It would be good if the methods being called have () behind it in the sequence diagram.

for Example: use printAndGetModule() instead of printAndGetModule. :)

jovanhuang commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,315 @@
+This feature allows user to add user-defined module components and its related weightage for an
+existing module.
+
+![Diagram](diagrams/addComponent.png)  
```
In the sequence diagram of addComponent, it would be good if it is shown where "command" is used for after it was returned to Component. (or at least  show that it is subsequently returned to ModuleInfo & Duke if "command" is used for other parts) :)

jovanhuang reviewed (commented):
Overall, looks good! :)

liping-eng committed:
Merge pull request #196 from liping-eng/branch-HelpGraduation

Add mc and grade (associated to module) to Module Storage.

nivikcivik committed:
tried fixing UG bugs at 3.3.3 and 3.3.5

nivikcivik committed:
changed will to may

nivikcivik committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp into nikhila-UserGuide

nivikcivik committed:
Merge pull request #197 from nivikcivik/nikhila-UserGuide

Nikhila user guide

hazelhedmine committed:
added delete task when module is deleted, fixed storage bug

hazelhedmine committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp into hazel-TaskManager

# Conflicts:
#	UniTracker Data/modules.txt
#	src/main/java/seedu/duke/ModuleInfo.java

hazelhedmine committed:
Merge pull request #199 from hazelhedmine/hazel-TaskManager

updated storage system and fixed delete task bug

KimIdeas8 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,315 @@
+- `Storage`: Write data to, and loads data to hard disk
+
+![Diagram](diagrams/MainArchitecture.png)
+
```
1. should ui be pointing to the other classes?
2. for a class name like 'helpGraduation', a suggestion can be to write in CamelCase. 
![image](https://user-images.githubusercontent.com/66380463/113196091-ac187b00-9295-11eb-9995-0acdef239d18.png)

KimIdeas8 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,315 @@
+user has to input a valid format for the __date__ and __time__ field (refer to section 3.3.1 of User
+Guide or the example above), and our application will convert it to a more readable format for the
+user. Using the example above,
+
```
It's nice to have numberings to improve formatting here :))

KimIdeas8 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,315 @@
+tasks in the `ArrayList` for the task type they chose will then be printed out, and the user has to
+input the index of the task they want to delete. The task they select will then be deleted from
+the `ArrayList`.
+
```
Same for improving formatting :))

KimIdeas8 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,315 @@
+3) `Date` <br> Example: `2021-03-21`
+4) `Time` <br> Example: `23:59`
+5) `Message` <br> Example: `Good job!`
+
```
cool to add `<br>` after each example so the next argument can fall on the next line :))
![image](https://user-images.githubusercontent.com/66380463/113199922-44b0fa00-929a-11eb-88a7-927be26ab3bb.png)

KimIdeas8 reviewed (commented):
Overall, nice DG over there! 👏💐🎂🥳💃

hazelhedmine committed:
updated user docs with task manager features

hazelhedmine committed:
Merge pull request #201 from hazelhedmine/hazel-UserGuide

updated user docs with task manager features

liping-eng committed:
Fix version 1 and 2 of capsimulator. Add conditions for valid CAP and set total number of MCs taken to range between 0 and 180.

liping-eng committed:
Merge pull request #202 from liping-eng/branch-HelpGraduation

Fix version 1 and 2 of capsimulator. Add conditions for valid CAP and…

liping-eng committed:
Fix SU grade to not count into totalMcs.

liping-eng committed:
Fix SU for version 2 of capSimulator. Add minor code changes.

liping-eng committed:
Merge pull request #203 from liping-eng/branch-HelpGraduation

Branch help graduation

SimBowen commented on docs/DeveloperGuide.md:
````diff
@@ -1,34 +1,315 @@
+Returning to module information menu...
+  ```
+Given below is the sequence diagram for the feature:  
+![Diagram](diagrams/addNewReview.png)  
````
Sequence diagram seems to be quite detailed, however may not follow the standard format. Do remember to also include the colon : at the front of the class.
![image](https://user-images.githubusercontent.com/60381634/113229868-63c78000-92ca-11eb-9d89-59731283cbf1.png)

Sequence diagram should not end in with the object at the bottom I think. Consider just leaving the dotted lines if the object is not destroyed?
![image](https://user-images.githubusercontent.com/60381634/113229813-4b576580-92ca-11eb-9b58-bd65a7083c40.png)

SimBowen replied:
Diagrams are also missing () for the methods.
![image](https://user-images.githubusercontent.com/60381634/113230871-8490d500-92cc-11eb-8856-5156b0d4b912.png)

Consider standardizing the sequence diagrams as the other diagrams in 4.01 and 4.05 have the brackets

![image](https://user-images.githubusercontent.com/60381634/113230849-7ba00380-92cc-11eb-99e8-5d665e23938a.png)

soepaingzaw commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,315 @@
+![Diagram](diagrams/addNewReview.png)  
+A general explanation of how this feature works:
+
+**Problem**: When the user wants to review a module, he/she should be able to type ___multiple lines
+of input___, as reviews are generally long and detailed.
+
+**Solution**: A `while` loop is used to take in multiple lines of input until the input contains
+the `/end` symbol. Anything typed after this `/end` symbol will be erased.
+
+**When overwriting this review with a new review**: The old review will be printed for user
+reference. The user will also be alerted that this action will delete his/her old review via a log.
+The log is printed out as a warning.
```
Appreciate this form of writing due to readability and comprehension. Maybe this can be improved by mentioning the classes involved or methods involved? Similar to what was written in section 4.05.
![image](https://user-images.githubusercontent.com/77471508/113230359-6d051c80-92cb-11eb-9685-c389ee52eacb.png)

SimBowen commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,315 @@
+5) `Message` <br> Example: `Good job!`
+
+Given below is the sequence diagram for the feature:  
+![Diagram](diagrams/addNewTask.png)
```
The sequence diagram is very detailed, as such it may be a bit lengthy and confusing? Consider splitting the blocks into separate diagrams and explaining them separately.

soepaingzaw commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,315 @@
+All task types (except `Task`, which contains arguments 1, 2 and 5) contain 5 arguments upon
+creation:
+
+1) `Module Code` <br> Example: `CS2113T`
+2) `Task Description` <br> Example: `Finish v2.0`
+3) `Date` <br> Example: `2021-03-21`
+4) `Time` <br> Example: `23:59`
+5) `Message` <br> Example: `Good job!`
```
Explanation here may not be intuitive due to format and structure of presentation viewed on the DG: 

![image](https://user-images.githubusercontent.com/77471508/113230896-970b0e80-92cc-11eb-8663-823ae2b7bd48.png)



Perhaps improving format can increase comprehension.

soepaingzaw commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,315 @@
+
+### 4.03 Add New Task
+
+This feature allows the user to add tasks of type `task`, `assignment`, `midterm`, `final exam` to a
+task list.
+
+All task types (except `Task`, which contains arguments 1, 2 and 5) contain 5 arguments upon
```
Best to standardize the capitalization of task types

SimBowen commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,315 @@
 # Developer Guide
 
-## Design & implementation
+## Table of Contents
```
The numberings of the content page do not line up
![image](https://user-images.githubusercontent.com/60381634/113231099-0da80c00-92cd-11eb-9235-22afd20350c0.png)
Additionally, the formatting of the sub sections are not standardized

SimBowen replied:
Numbering in the sections themselves are ok.

soepaingzaw commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,315 @@
+### 4.05 Add Zoom Link
+
+Given below is the sequence diagram for the feature:  
+![Diagram](diagrams/addZoomLink.png)
```
Good Sequence diagram. Perhaps it is alright to end the sequence diagram at the bottom with just lifelines? Maybe there is no need to end it with the object "boxes"

![image](https://user-images.githubusercontent.com/77471508/113231841-a9864780-92ce-11eb-9ea1-e582d0a8d1ce.png)

soepaingzaw reviewed (commented):
Overall a comprehensive DG with aptly used sequence diagrams. Understandably, certain components are still in process. Further refinement will set you guys in for the final submission!

motorpeople101 committed:
merge changes

motorpeople101 committed:
add changes

motorpeople101 committed:
fixing merge conflicts

motorpeople101 committed:
add changes suggested by peers from feedback

BlubberMonster committed:
Merge pull request #204 from BlubberMonster/master

add changes suggested by peers from feedback

hazelhedmine committed:
updated introduction section for dg

hazelhedmine committed:
updated dg with sequence diagrams

hazelhedmine committed:
Merge pull request #206 from hazelhedmine/hazel-UserGuide

updated DG for introduction and task manager methods

nivikcivik committed:
removed add/view mod description from UG

no longer using that command

liping-eng committed:
Fix Storage for module.txt by removing space at end of  " -- end of module -- " as text file seem to keep auto omitting the space behind. Update component to similar interface, menu style.

liping-eng committed:
Merge pull request #209 from liping-eng/branch-UserGuide

Fix Storage for module.txt by removing space at end of  " -- end of m…

nivikcivik committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp into nikhila-UserGuide

nivikcivik committed:
Merge pull request #208 from nivikcivik/nikhila-UserGuide

removed add/view mod description from UG

liping-eng committed:
Update user guide on components, helpgraduation features.

liping-eng committed:
Merge pull request #210 from liping-eng/branch-UserGuide

Update user guide on components, helpgraduation features.

liping-eng committed:
no message

liping-eng committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp into branch-ModuleComponents

BlubberMonster committed:
Set theme jekyll-theme-cayman

liping-eng committed:
add storage for component.

liping-eng committed:
Merge pull request #246 from liping-eng/branch-ModuleComponents

Branch module components

liping-eng committed:
Checkstyle + confirm no build errors after storage is fixed for components.

liping-eng committed:
Merge pull request #247 from liping-eng/branch-ModuleComponents

Checkstyle + confirm no build errors after storage is fixed for compo…

liping-eng committed:
Fix ArrayOutOfBoundsIndex exception, empty hashtable flaw in Component.

liping-eng committed:
Merge pull request #248 from liping-eng/branch-ModuleComponents

Fix ArrayOutOfBoundsIndex exception, empty hashtable flaw in Component.

hazelhedmine committed:
fixed task manager bugs from mock pe

hazelhedmine committed:
Merge pull request #249 from hazelhedmine/hazel-TaskManager

fixed task manager bugs from mock pe

hazelhedmine committed:
brushed up intro, setting up, faq and task manager sections

hazelhedmine committed:
fixed bugs

hazelhedmine committed:
Merge pull request #252 from hazelhedmine/hazel-TaskManager

UG update and bug fix

nivikcivik committed:
nikhila - 1st round of fixing bugs

viewAModule ends when empty list, module name/description is a must, module name must be alphanumeric, error msg only shown once now.
UG bugs: dotted line after review is removed, shown how to enter ModInfo menu and for other actions

nivikcivik committed:
checkStyle fixes

nivikcivik committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp into nikhila-moduleInfoMenu

# Conflicts:
#	src/main/java/seedu/duke/ModuleInfo.java

nivikcivik committed:
Merge pull request #254 from nivikcivik/nikhila-moduleInfoMenu

nikhila - 1st round of bug fixing

motorpeople101 committed:
add changes to fix bugs

BlubberMonster committed:
Merge pull request #255 from BlubberMonster/master

add changes to fix bugs

motorpeople101 committed:
add changes to fix another bug in zoomlink delete

motorpeople101 committed:
add changes to fix bugs in delete zoomlinks

motorpeople101 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

BlubberMonster committed:
Merge pull request #256 from BlubberMonster/master

fix bugs in delete zoomlink

motorpeople101 committed:
add extra details to ugdg

motorpeople101 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

BlubberMonster committed:
Merge pull request #257 from BlubberMonster/master

add changes to ugdg

nivikcivik committed:
code after meeting

packaging, Storage bug fixes, Ui

nivikcivik committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp into nikhila-mainMenu

# Conflicts:
#	src/main/java/seedu/duke/features/link/Links.java

nivikcivik committed:
checkStyle fixes

nivikcivik committed:
Merge pull request #260 from nivikcivik/nikhila-mainMenu

after tP meeting - Storage bug fixes & repackaging

motorpeople101 committed:
add logging to text file

motorpeople101 committed:
add logging to text file

BlubberMonster committed:
Merge pull request #261 from BlubberMonster/master

add logging to text file

liping-eng committed:
fix component empty.

liping-eng committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp into branch-ModuleComponents

liping-eng committed:
Merge pull request #262 from liping-eng/branch-ModuleComponents

fix component empty.

liping-eng committed:
fix wrong cap. cap is set to round at 2d.p.

liping-eng committed:
Merge pull request #263 from liping-eng/branch-ModuleComponents

fix wrong cap. cap is set to round at 2d.p.

liping-eng committed:
fix double line in helpgraduation.

liping-eng committed:
Merge pull request #264 from liping-eng/branch-ModuleComponents

fix double line in helpgraduation.

liping-eng committed:
fix UG to note only accept integer. check if name is integer to prevent user inputting it as weightage.

liping-eng committed:
no message

liping-eng committed:
Merge pull request #265 from liping-eng/branch-ModuleComponents

Branch module components

liping-eng committed:
no message

liping-eng committed:
Merge pull request #266 from liping-eng/branch-ModuleComponents

no message

motorpeople101 committed:
add diagrams and logging to dg

BlubberMonster committed:
Merge pull request #267 from BlubberMonster/master

add diagrams and logging to dg

liping-eng committed:
Fix cap to print up to 2dp. Add addcapandmcs to DG.

liping-eng committed:
Merge pull request #268 from liping-eng/branch-ModuleComponents

Fix cap to print up to 2dp. Add addCapAndMcs to DG.

motorpeople101 committed:
add changes to dg

motorpeople101 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

motorpeople101 committed:
added exception to deal with link loading eror and update ug

BlubberMonster committed:
Merge pull request #269 from BlubberMonster/master

add exception to deal with invalid link loading error and update ug

hazelhedmine committed:
fixed some bugs and added user stories

hazelhedmine committed:
Merge pull request #270 from hazelhedmine/hazel-TaskManager

fixed some bugs and added user stories

hazelhedmine committed:
added logging for task manager features

hazelhedmine committed:
Merge pull request #273 from hazelhedmine/hazel-TaskManager

added logging for task manager features

motorpeople101 committed:
update readme

motorpeople101 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

motorpeople101 committed:
fix bugs and add javadoc

motorpeople101 committed:
fix checkstyle

BlubberMonster committed:
Merge pull request #274 from BlubberMonster/master

fix bugs and add javadoc

liping-eng committed:
draft PPP.

liping-eng committed:
Merge pull request #275 from liping-eng/branch-ModuleComponents

draft PPP.

motorpeople101 committed:
fix bugs in links and update ug

motorpeople101 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

BlubberMonster committed:
Merge pull request #276 from BlubberMonster/master

fix bugs in links and update ug

motorpeople101 committed:
update ug

BlubberMonster committed:
Merge pull request #277 from BlubberMonster/master

update ug

hazelhedmine committed:
added PPP for hazel and fixed checkstyle

hazelhedmine committed:
Merge pull request #282 from hazelhedmine/hazel-AboutUs

added PPP for hazel and fixed checkstyle

nivikcivik committed:
nikhila - DG fixes

Fixed old sequence diagram, added one more sequence diagram, edited code for consistency with sequence diagrams;
Fixed some bugs for deleteReview

nivikcivik committed:
checkStyle fixes

nivikcivik committed:
Merge pull request #285 from nivikcivik/nikhila-DeveloperGuide

nikhila - DG fixes & Bug Fixes

nivikcivik committed:
big fixes - yes or no features of addNewReview and deleteReview

nivikcivik committed:
checkStyle fixes

nivikcivik committed:
Merge pull request #288 from nivikcivik/nikhila-moduleInfoMenu

bug fixes - yes or no features of addNewReview and deleteReview

motorpeople101 committed:
fix bugs where only one zoom link could be tagged per mod, and zoom link still having a module even after module has been deleted

motorpeople101 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

motorpeople101 committed:
fix checkstyle error

motorpeople101 committed:
fix checkstyle

BlubberMonster committed:
Merge pull request #290 from BlubberMonster/master

fix bugs

hazelhedmine committed:
worked on dg

hazelhedmine committed:
Merge pull request #293 from hazelhedmine/hazel-UserGuide

worked on dg

nivikcivik committed:
nikhila - worked on documentation

PPP, DG - manual testing, Architecture diagram

nivikcivik committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp into nikhila-DeveloperGuide

# Conflicts:
#	docs/DeveloperGuide.md
#	docs/diagrams/featuresArchitecture.puml

nivikcivik committed:
Merge pull request #296 from nivikcivik/nikhila-DeveloperGuide

Nikhila developer guide

liping-eng committed:
no message

liping-eng committed:
Merge pull request #297 from liping-eng/branch-ModuleComponents

no message

liping-eng committed:
no message

liping-eng committed:
Merge pull request #298 from liping-eng/master

no message

liping-eng committed:
uml diagram for features.

liping-eng committed:
Merge pull request #299 from liping-eng/master

uml diagram for features.

motorpeople101 committed:
update dg

BlubberMonster committed:
Merge pull request #300 from BlubberMonster/master

update dg

motorpeople101 committed:
add PPP draft

motorpeople101 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

BlubberMonster committed:
Merge pull request #302 from BlubberMonster/master

add PPP draft

motorpeople101 committed:
fix visual bug in ppp draft

motorpeople101 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

BlubberMonster committed:
Merge pull request #303 from BlubberMonster/master

fix visual bug in ppp draft for yong han

motorpeople101 committed:
fix visual bug in ppp draft for yh

motorpeople101 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

BlubberMonster committed:
Merge pull request #304 from BlubberMonster/master

fix visual bug in ppp draft for yh

motorpeople101 committed:
fix visual bug in the dg manual testing part

motorpeople101 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

BlubberMonster committed:
Merge pull request #305 from BlubberMonster/master

fix visual bug in dg manual testing part

motorpeople101 committed:
fix visual bug again in dg manual testing part

motorpeople101 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

BlubberMonster committed:
Merge pull request #306 from BlubberMonster/master

fix visual bug in dg manual testing part

motorpeople101 committed:
fix visual bug in dg manual testing

motorpeople101 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

BlubberMonster committed:
Merge pull request #307 from BlubberMonster/master

fix visual bug in dg manual testing part

motorpeople101 committed:
add javadocs to the links package

motorpeople101 committed:
fix checkstyle

motorpeople101 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

BlubberMonster committed:
Merge pull request #309 from BlubberMonster/master

add javadocs to the links package

liping-eng committed:
make use of Enum class. Show SU grade in simulator version 1 and 2.

liping-eng committed:
add enumeration title.

liping-eng committed:
Merge branch 'master' into master

liping-eng committed:
Merge pull request #312 from liping-eng/master

make use of Enum and fix SU on vers 1 and 2 of simulator

liping-eng committed:
add logger to classes under capsimulator package and component class.

liping-eng committed:
Merge pull request #314 from liping-eng/master

add logger to classes under capsimulator package and component class.

hazelhedmine committed:
added javadoc for task manager, tasks storage and Ui

hazelhedmine committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp into hazel-TaskManager

hazelhedmine committed:
Fixed checkstyle

hazelhedmine committed:
Merge pull request #315 from hazelhedmine/hazel-TaskManager

added javadoc for task manager parts

liping-eng committed:
Fix userguide under ModuleInfo.

liping-eng committed:
Merge pull request #316 from liping-eng/master

Fix userguide under ModuleInfo.

motorpeople101 committed:
fixes bugs where user adds zoom link that is not a zoom link

motorpeople101 committed:
update dg diagram in addZoomLink

motorpeople101 committed:
fix bug with initialise links method

motorpeople101 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

BlubberMonster committed:
Merge pull request #317 from BlubberMonster/master

fix bugs when zoomLink is invalid and initialiseLinks logic

liping-eng committed:
updated user guide and print methods.

liping-eng committed:
Merge pull request #318 from liping-eng/master

updated user guide and print methods.

nivikcivik committed:
committing to pull new changes

nivikcivik committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp into nikhila-UserGuide

liping-eng committed:
update userguide.

liping-eng committed:
Merge pull request #319 from liping-eng/master

update userguide.

liping-eng committed:
update userguide

liping-eng committed:
Merge pull request #320 from liping-eng/master

update userguide

nivikcivik committed:
committing before pulling latest code

nivikcivik committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp into nikhila-UserGuide

nivikcivik committed:
added "go to xyz" links at each feature/section

nivikcivik committed:
Merge pull request #323 from nivikcivik/nikhila-UserGuide

Nikhila user guide

liping-eng committed:
update developer guide.

liping-eng committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

motorpeople101 committed:
changed naming of methods

motorpeople101 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

nivikcivik committed:
Added one more section to DG and updated TOC

nivikcivik committed:
added one more line to Section 4.06 View All Reviews

nivikcivik committed:
Merge pull request #338 from nivikcivik/nikhila-moduleInfoMenu

added viewAllReviews section to DG

nivikcivik committed:
fixed review bugs

("END" is accepted and anything after 'end' noted in UG)

nivikcivik committed:
fixed review bug for delimiter

nivikcivik committed:
Merge pull request #339 from nivikcivik/nikhila-moduleInfoMenu

fixed review bugs

nivikcivik committed:
PPP - first draft

nivikcivik committed:
edited PPP & addNewReview diagram

nivikcivik committed:
Merge pull request #343 from nivikcivik/nikhila-AboutUs

Nikhila 1st PPP draft

motorpeople101 committed:
fix bug with  missing "returning to menu" at links package and also standardise capitalisation of words

motorpeople101 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

BlubberMonster committed:
Merge pull request #345 from BlubberMonster/master

add fix to missing "Returning to menu " in links package and standardise capitalisation in words

hazelhedmine committed:
updated PPP and task manager bug

hazelhedmine committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp into hazel-TaskManager

hazelhedmine committed:
PPP bug

hazelhedmine committed:
Merge pull request #346 from hazelhedmine/hazel-TaskManager

Hazel task manager

liping-eng committed:
fix bugs.

liping-eng committed:
fix bugs. resolve merge conflict in DG due to view all reviews.

liping-eng committed:
Merge pull request #347 from liping-eng/master

fix bugs, resolve dg conflict

motorpeople101 committed:
add minor change to ui

hazelhedmine committed:
fixed ug bugs

motorpeople101 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

hazelhedmine committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp into hazel-UserGuide

BlubberMonster committed:
Merge pull request #348 from BlubberMonster/master

fix minor ui bug

hazelhedmine committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp into hazel-UserGuide

hazelhedmine committed:
fixed DG bugs, fixed sizing of some diagrams and added figure numbers

hazelhedmine committed:
Merge pull request #349 from hazelhedmine/hazel-UserGuide

edited DG

motorpeople101 committed:
update ppp

motorpeople101 committed:
fix bugs in ugdg

BlubberMonster committed:
Merge pull request #350 from BlubberMonster/master

add details to ppp, update ugdg

liping-eng committed:
fix DG info and diagrams.

liping-eng committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

# Conflicts:
#	docs/DeveloperGuide.md
#	docs/diagrams/capCalculator/addCapAndMcs.png
#	docs/diagrams/moduleInfo/addComponent.png
#	src/main/java/seedu/duke/ui/Ui.java

liping-eng committed:
Merge pull request #351 from liping-eng/master

resolve merge conflict, fix DG and diagrams

liping-eng committed:
fix resolution.

liping-eng committed:
Merge pull request #352 from liping-eng/master

fix resolution.

liping-eng committed:
fix 125% resolution.

liping-eng committed:
Merge pull request #353 from liping-eng/master

fix 125% resolution.

liping-eng committed:
no message

liping-eng committed:
Merge pull request #354 from liping-eng/master

no message

motorpeople101 committed:
update AboutUs

BlubberMonster committed:
Merge pull request #355 from BlubberMonster/master

update AboutUs

motorpeople101 committed:
fix AboutUs

motorpeople101 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp

BlubberMonster committed:
Merge pull request #356 from BlubberMonster/master

fix AboutUs

liping-eng committed:
add JavaDocs

liping-eng committed:
Merge pull request #357 from liping-eng/master

add JavaDocs

liping-eng committed:
update userguide on cap simulator.

liping-eng committed:
Merge pull request #358 from liping-eng/master

update userguide on cap simulator.

liping-eng committed:
Complete PPP.

liping-eng committed:
Merge pull request #359 from liping-eng/master

Complete PPP.

liping-eng committed:
update PPP.

liping-eng committed:
Merge pull request #360 from liping-eng/master

update PPP.

liping-eng committed:
fix checkstylemain in JavaDoc for Component class.

liping-eng committed:
Merge pull request #361 from liping-eng/master

fix checkstylemain in JavaDoc for Component class.

nivikcivik committed:
removed warning from DG + JavaDoc

nivikcivik committed:
Merge pull request #364 from nivikcivik/nikhila-UserGuide

removed warning from DG + JavaDoc

nivikcivik committed:
updated PPP links

liping-eng committed:
update PPP and fix bug in moduleInfo.

liping-eng committed:
Merge pull request #365 from liping-eng/master

update PPP and fix bug in moduleInfo.

liping-eng committed:
make more OOP, fix acadamicRecords, ModuleGradeEnum prints to Ui class.

liping-eng committed:
Merge pull request #366 from liping-eng/master

make more OOP, fix acadamicRecords, ModuleGradeEnum prints to Ui class.

liping-eng committed:
fix up limit for total mcs.

liping-eng committed:
fix mc overflow feature flaw.

liping-eng committed:
Merge pull request #367 from liping-eng/master

mc overflow fixed.

nivikcivik committed:
ugdg final edits

nivikcivik committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-4/tp into nikhila-UserGuide

nivikcivik committed:
Merge pull request #369 from nivikcivik/nikhila-UserGuide

UGDG final edits

liping-eng committed:
fix bug.

liping-eng committed:
Merge pull request #370 from liping-eng/master

fix bug.
